app/assets/javascripts/components/overview/campaign_button.jsx

```diff
@@ -29,13 +30,24 @@ const CampaignButton = createReactClass({
   },
 
   removeCampaign(campaignId) {
+    this.addDeletedCampaign(campaignId);
```

Comment:
@ragesoss I've been struggling with this as I was trying to perform these changes in the reducer -- without sucess. Right now is partially working: I would need to send the allCampaigns that remain available to the state the app doesn't show all the campaigns in the next interaction. I will research on that tomorrow (suggestions?). Also, would you create these new methods to add and delete campaigns from the state or would you just add this actions to addCampaign/removeCampaign without creating specific methods? What do you think is a best practice?

Reply:
I would say that you want the redux state to hold two things: the campaigns for this course, and the set of all campaigns. Then, for displaying a list of possible campaigns to add, use a selector to remove the course's campaigns from the full set. If the server returns the list of a course's campaigns when you add or remove, then it can be the same redux action fetching course campaigns initially, adding one, or removing one... the full list of course campaigns comes fresh from the server in each case..

Reply:
For me to see if I understood you correctly: Would you add e.g. a second argument to the action (fetchAllCampaigns) with all the course campaigns (e.g. fetchAllCampaigns(campaignId, courseCampaigns)) and then in the success case of the action, compare the two arrays and send as data only the campaigns difference?

Reply:
Sorry, it would be "fetchAllCampaigns(courseCampaigns)" as we are not sending the campaignId to fetch campaigns.